### PR TITLE
Development

### PR DESCRIPTION
**Pull request by pacyL2K19**
# The aim of this PR
This project was to create a bubble sort method in Ruby
We used Yield to execute additional instruction on method invocation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,41 @@
...
+
+## PREREQUISITE
+
+- Install at least one code editor 
+- Install Ruby to exectute the project
+
+## GETTING STARTED
+
+- clone the repository `$ git clone https://github.com/pacyL2K19/bubble_sort.git`
+- Run the command ruby bubble_sort.rb
```
Comment (SafaErden) [inline README.md]: - [x] Don't forget to remind users to change the directory after cloning the app.

Comment (SafaErden) [inline README.md]: - [x] You don't have a bubble_sort.rb file to execute. Instead you have 'bubble_sort_by' and 'bubble_sort_secondary' files. Please kindly update your instructions to execute the file properly.

File: bubble_sort_secondary.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+			if curr_value > next_value
+				arr[i] = next_value
+				arr[i + 1] = curr_value
+				exchanged = true
+			end
+		end
+		sorted = !exchanged ? true : false;
+		p arr
+	end
+	# p arr
```
Comment (SafaErden) [inline bubble_sort_secondary.rb]: - [x] Please remove commented code from your application.

File: bubble_sort_by.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    swapped = false
+    n.times do |i|
+      if yield(arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap = true
+      end
+    end
+    break if swapped == false
+  end
+  p arr
```
Comment (SafaErden) [inline bubble_sort_by.rb]: - [x] Please just return results instead of printing them.

File: bubble_sort_secondary.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+			curr_value = arr[i]
+			next_value = arr[i + 1]
+			if curr_value > next_value
+				arr[i] = next_value
+				arr[i + 1] = curr_value
+				exchanged = true
+			end
+		end
+		sorted = !exchanged ? true : false;
+		p arr
```
Comment (SafaErden) [inline bubble_sort_secondary.rb]: - [x] Please just return results instead of printing them.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,41 @@
-# bubble_sort
-The bubble sort algorithm made in Ruby 
+# Bubble_sort
+
+> This is the first project in the Ruby module 
+
+## PREREQUISITE
+
+- Install at least one code editor 
+- Install Ruby to exectute the project
```
Comment (SafaErden) [inline README.md]: - [x] Please kindly fix the spelling error (execute).

**Review by SafaErden — CHANGES_REQUESTED** [review]
## Status: Require changes ♻️

**Hi TEAM, your project is awesome. 👐 However, we need a little fix before we can proceed. Kindly find comments below ⤵**

##

**REQUIRED CHANGES**

- [x] Listed below, please kindly fix them.

##

**OPTIONAL CHANGES**

- [x] Instead of creating two separate files for the two separate sorting methods, please put them into one file named 'bubble_sort.rb' .

##

**NOTES**

Friendly reminder: Remember it's important to have a professional repo so you can increase your job opportunities. Please take a step back and read the lesson [here.](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)

##

**SUMMARY**

As I mentioned above, your project is almost done. You have some minor issues to solve. Please be an optimist and do not forget that the main purpose is to learn and practice. Do not let the requested changes discourage you 👍

##

*Kindly make the changes and submit another code review when you are ready💪*

Questions or comments? Reach me on [Twitter](https://twitter.com/safaerden) or [LinkedIn](https://www.linkedin.com/in/safaerden/) 🗨️

Happy coding_ 👨‍💻 

**Safa ERDEN**

**Comment by mimipeshy** [comment]
## STATUS: APPROVED 🥳 🍾 

Hey team 👋 👋 , good job on working on all required changes 👏  👏 .Your project is approved, go ahead and merge. All the best in the next project 👯 

Regards
Peris Ndanu

**Comment by pacyL2K19** [comment]
Thank you so much

**Pull request merged**